packages/webapp/src/components/Animals/AddAnimalsFormCard/AddAnimalsFormCard.tsx

```diff
@@ -0,0 +1,144 @@
+  count?: number;
+  createIndividualProfiles?: boolean;
+  group?: string;
+  batch?: string;
+};
+
+type AddAnimalsFormCardProps = AnimalTypeSelectProps &
+  AnimalBreedSelectProps & {
+    sexDetailsOptions: SexDetailsType;
+    onIndividualProfilesCheck?: (isChecked: boolean) => void;
+    onRemoveButtonClick?: (e: React.MouseEvent<HTMLButtonElement>) => void;
+    showRemoveButton?: boolean;
+    isActive?: boolean;
+  };
+
+export default function AddAnimalsFormCard({
+  typeOptions,
+  breedOptions,
+  sexDetailsOptions,
+  onTypeChange,
+  onIndividualProfilesCheck,
+  showRemoveButton,
+  onRemoveButtonClick,
+  isActive,
+}: AddAnimalsFormCardProps) {
+  const { control, watch, register } = useFormContext<FormFields>();
+  const { t } = useTranslation();
+  const watchAnimalCount = watch('count') || 0;
+  const watchAnimalType = watch('type');
+  const shouldCreateIndividualProfiles = watch('createIndividualProfiles');
+
+  return (
+    <Card className={styles.form} isActive={isActive}>
+      <div className={styles.formHeader}>
+        <Text>{t('ADD_ANIMAL.ADD_TO_INVENTORY')}</Text>
+        {showRemoveButton && (
+          <TextButton className={styles.removeBtn} onClick={onRemoveButtonClick}>
+            {' '}
+            <XIcon /> {t('common:REMOVE')}
+          </TextButton>
+        )}
+      </div>
+      <AnimalTypeSelect
+        name="type"
+        control={control}
+        typeOptions={typeOptions}
+        onTypeChange={onTypeChange}
+      />
+      <AnimalBreedSelect
+        name="breed"
+        control={control}
+        breedOptions={breedOptions}
+        showNoTypeSelectedMessage={!watchAnimalType}
+      />
+
+      <div className={styles.countAndSexDetailsWrapper}>
+        <NumberInput
+          name="count"
+          control={control}
+          defaultValue={0}
+          label={t('common:COUNT')}
+          className={styles.countInput}
+          allowDecimal={false}
+          showStepper
+        />
+        <Controller
+          name="sexDetails"
+          control={control}
+          render={({ field }) => (
+            <SexDetails
+              initialDetails={sexDetailsOptions}
+              maxCount={watchAnimalCount}
+              onConfirm={(details) => field.onChange(details)}
+            />
+          )}
+        />
+      </div>
+      <Checkbox
+        label={t('ADD_ANIMAL.CREATE_INDIVIDUAL_PROFILES')}
+        tooltipContent={t('ADD_ANIMAL.CREATE_INDIVIDUAL_PROFILES_TOOLTIP')}
+        hookFormRegister={register('createIndividualProfiles')}
+        onChange={(e) => onIndividualProfilesCheck?.((e.target as HTMLInputElement).checked)}
+      />
+      {shouldCreateIndividualProfiles ? (
+        // @ts-ignore
+        <Input
+          label={t('ADD_ANIMAL.GROUP_NAME')}
+          optional
+          placeholder={t('ADD_ANIMAL.GROUP_NAME_PLACEHOLDER')}
+          hookFormRegister={register('group')}
+        />
+      ) : (
+        // @ts-ignore
+        <Input
```

Comment:
This` Input` is handling text inputs, number inputs, date pickers, and passwords inputs all in one. It might be good to extract those out into different components and also use typescript to avoid ts-ignore.